src/compiler/binder.ts

```diff
@@ -1945,6 +1951,9 @@ namespace ts {
                     // is an ES6 node.
                     transformFlags = TransformFlags.AssertES6;
                 }
+                if (subtreeFlags & TransformFlags.ContainsLexicalThisInComputedPropertyName) {

```

Comment:
Could you add comment about the discussion you have with @rbuckton why you convert this to "ContainsLexicalThis"